runtime/lua/vim/fs.lua

```diff
@@ -248,7 +248,7 @@ end
 --- ```
 ---
 ---@since 10
----@param names (string|string[]|fun(name: string, path: string): boolean) Names of the items to find.
+---@param names (string|string[]|vim.lpeg.Pattern|vim.lpeg.Pattern[]|fun(name: string, path: string): boolean) Names of the items to find.
```

Comment:
rather than continue to add more overloads, this may need to be a kwargs arg
```suggestion
---@param names (string|string[]|table|fun(name: string, path: string): boolean) Names of the items to find.
```

and currently, the kwargs arg can support `{ glob: string }`. the users can do `vim.fs.find({ glob: '*/*.txt' })`. WDYT @gpanders

Reply:
Yeah that's much cleaner. The only reason I went with this approach is that I was already thinking about how it could integrate with `vim.fs.root()` and in `root_markers`

We could end up with something like:

```lua
root_markers = { { { glob = "*pattern" }, "lua_ls" }, ".git" }
```

So I thought a cleaner approach might be:

```
root_markers = { { vim.glob.to_lpeg("*pattern"), "lua_ls" }, ".git" }
```

That said, I haven't really looked into how many configurations would need this kind of `root_markers`.

Reply:
I also thought about adding a flag to  `opts`, like `{ glob = true }`, but I don't know how it could be integrated with `root_markers`, unless everything is treated as a glob in `root_markers`.